### ISSUE TITLE
After Upgrading service fabric to  6.4.637.9590 getting System.UpgradeOrchestrationService' reported Warning for property 'ManifestVersionMismatch'. Mismatched cluster manifest versions: 1 0. Please perform a config upgrade to sync manifest state. --

### ISSUE THREAD

**Issue by rock11rock22** [Azure/service-fabric-issues#1451]
1. 

'System.UpgradeOrchestrationService' reported Warning for property 'ManifestVersionMismatch'. Mismatched cluster manifest versions: 1 0. Please perform a config upgrade to sync manifest state.
--

**Comment by rock11rock22** [comment]
is anyone know the solution of this issue ?

**Comment by darran1971** [comment]
I've got this warning after upgrading to 6.4.637.9590 also. It appeared a couple of days after the actual upgrade.

**Comment by rock11rock22** [comment]
are you able to resolve the issue ?

**Comment by rock11rock22** [comment]
> I've got this warning after upgrading to 6.4.637.9590 also. It appeared a couple of days after the actual upgrade.
> 
> ![mismatch](https://user-images.githubusercontent.com/40686134/52939793-e7c58480-335c-11e9-9b3d-0ac0d1628082.png)

are you able to resolve the issue ?

**Comment by wuping2004** [comment]
![outlook_2019-02-22_09-24-08](https://user-images.githubusercontent.com/17618809/53213378-a813db00-3683-11e9-98f0-7b7f9d1991b1.png)

I also met this issue, is there any mitigation steps, thanks

**Comment by wuping2004** [comment]
I notice the TTL is two days, is warning message gone automatically after two days?

**Comment by darran1971** [comment]
No, the warning message returns after the TTL expires. I have tried updating the config, but the error returns, with the current version incremented in the error message.

**Comment by Adebeer** [comment]
Noticed same issue today on my cluster aswell - upgrade performed just fine few days ago.

**Comment by Adebeer** [comment]
Can we please get an update on this issue? Issue remains - we've done several config upgrades to add new nodes.

Unhealthy event: SourceId='System.UpgradeOrchestrationService', Property='ManifestVersionMismatch', HealthState='Warning', ConsiderWarningAsError=false.
Mismatched cluster manifest versions: 21 13. Please perform a config upgrade to sync manifest state.

**Comment by jkochhar** [comment]
@rock11rock22 Did you attempt a 'cluster manifest' upgrade directly on these clusters? The warning is intended to make sure the state in UOS matches with the current cluster manifest else config upgrades can fail. 
Can you share the cluster manifest version?

**Comment by rock11rock22** [comment]
> @rock11rock22 Did you attempt a 'cluster manifest' upgrade directly on these clusters? The warning is intended to make sure the state in UOS matches with the current cluster manifest else config upgrades can fail.
> Can you share the cluster manifest version?

No, I just upgraded cluster using Start-ServiceFabricClusterUpgrade -Code -CodePackageVersion 6.4.637.9590 -Monitored -FailureAction Rollback

The warning appeared for couple of days and now its disappeared. 
We have another cluster where we did upgraded 2 days back and now those new upgraded cluster started showing warnings.

Unhealthy event: SourceId='System.UpgradeOrchestrationService', Property='ManifestVersionMismatch', HealthState='Warning', ConsiderWarningAsError=false. Mismatched cluster manifest versions: 1.35.0 0. Please perform a config upgrade to sync manifest state.

Manifest File :
<ClusterManifest xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="xxxxxxxxxxxx" Version="1.35.0" Description="Updated Manifest" xmlns="http://schemas.microsoft.com/2011/01/fabric">

**Comment by Adebeer** [comment]
Thanks @rock11rock22: Cluster manifest upgrades are not supported/recommended by MS, right? I've built a brand new cluster and had same issue - never done manifest upgrade - only config updates. I also did a dummy upgrade (i.e. only incremented version number) but the warning unfortunately remains.

**Comment by maburlik** [comment]
You may be able to manually mitigate the issue by refreshing the node configuration on that particular node using the new manifest.
1. Disable the node.
2. Stop FabricHost service.
3. Run
New-ServiceFabricNodeConfiguration -ClusterManifestPath c:\temp\clustermanifest.33.xml

From what I've gathered people are seeing this after running a code upgrade? Is there anything special about the nodes on which the manifest is out of sync? Are they added nodes, or are they VMs which had been reverted to an old snapshot?

Please send us the Fabric etl logs from the period of the upgrade that induced the manifest desynchronization, from the FabricLogRoot of the nodes running UOS.

**Comment by jkochhar** [comment]
@Adebeer Please share the cluster manifest version (or your clustermanifest.xml) and the output of this API Get-ServiceFabricUpgradeOrchestrationServiceState. You can mail me at jkochhar AT Microsoft DOT com

**Comment by Adebeer** [comment]
@jkochhar Good news - I no longer seem to get this. New cluster on 6.4.637.9590. 
I just added 5 new nodes to this cluster and didn't experience the warning.

**Comment by Adebeer** [comment]
Thanks @jkochhar - I emailed you json/xml files today for your inspection.

The warning re-appeared today after adding the 5 nodes yesterday. 
Given that I didn't have the warning prior to adding the 5 nodes, I suspect the warning may go away after couple of hours/days. I say may, because I re-created this cluster from scratch - for the previous cluster, I'm pretty sure I saw this warning for more than 1-2 weeks.

FYI: For this cluster that I recreated - I've only added nodes to it via config upgrades - no other operations performed. In all cases, I waited for configuration to complete (even the auto-initiated ones that normally commence after adding nodes to add/adjust Seed Nodes)

**Comment by Adebeer** [comment]
Update for everyone - based on assistance I got from @jkochhar, I've been able to get rid of warning.

Simply, identify which node is running the primary instance of UpgradeOrchestrationService - and restart that node. That'll essentially fail over primary to one of other secondaries. In my case this got rid of the warning and it didn't return next day. 

If you don't want to restart whole node, you can try just restarting the primary instance, although I haven't tried this myself. Presumably you can do this with `Restart-ServiceFabricReplica` powershell commandlet

**Comment by garrmark** [comment]
This still appears to happen even in 6.5.658.9590. I can confirm that restarting the primary instance, as @Adebeer mentions above, resolves the warning.

**Comment by kaisxu** [comment]
Hit this error in 6.5.664.9590. @Adebeer it works for me too.

**Comment by 11-lein** [comment]
With Code Version 7.0.457.9590 on the on-premise cluster, restarting the master-node (reboot) nor restarting the fabric:/System/UpgradeOrchestrationService-service didn't resolve the warning.
